### PR TITLE
docs: Add sample-configuration for TLS client options

### PR DESCRIPTION
**Pull request by srebhan**
- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

Add a sample-configuration for the TLS client-config.

### CODE REVIEW

File: plugins/common/tls/config.go [hunk 1]
```diff
@@ -3,13 +3,17 @@ package tls
...
+	_ "embed"
 	"fmt"
 	"os"
 	"strings"
 
 	"github.com/influxdata/telegraf/internal/choice"
 )
 
+//go:embed client.conf
+var clientConfig string
```
Comment (powersj) [inline plugins/common/tls/config.go:15]: Hmm not sure I like having `ClientConfig` and `clientConfig`, how about using sampleConfig here as well like we do in other plugins?

Comment (srebhan) [reply plugins/common/tls/config.go:15]: Yeah that's confusing I agree. Simply `sampleConfig` is probably not enough as that file also contains a `ServerConfig` with different fields... How about `sampleClientConfig`?

Comment (powersj) [reply plugins/common/tls/config.go:15]: sounds good to me!

**Review by Hipska — APPROVED** [review]

**Comment by srebhan** [comment]
After some more thoughts this approach has several drawbacks:
1. With plugins ending with a map definition in TOML e.g. `[inpus.modbus.workarounds]` this approach will not be possible and will generate malformed (and/or confusing) TOML files.
2. The `sample.conf` files do no longer match the output of the `SampleConfig` function call.
3. Inserting into a config structure is very limited as you can only add stuff at the beginning or end.
4. There is no way to comment/uncomment an embedded config in a good way.

Given all those limitation (especially 1 and 2) I'm closing this PR as we need to come up with something more flexible.

**Comment by Hipska** [comment]
1 could be solved by not including it for those kind of plugins, but I agree to close and find a better solution.

**Comment by srebhan** [comment]
> Maybe let sample.conf be a golang template where you include the tls.conf?

@Hipska that would be one way. In any way we should generate a `sample.conf` that is the rendered result and check that in to the repo. So this calls for another "generator" running with `make docs`...

**Comment by Hipska** [comment]
I don't see the need for that, why would that rendered sample.conf need to be checked in the repo? (It only needs to be rendered in the readme, but that is currently already the case..

File: plugins/common/tls/config.go (hunk 1 above)
Comment (Hipska) [reply plugins/common/tls/config.go:15]: What about splitting this code also into a `client.go` and `server.go`?

**Comment by Hipska** [comment]
I do like the tls client.conf sample, that should be added to the repo anyway.

**Comment by powersj** [comment]
@srebhan close this PR in favor of https://github.com/influxdata/telegraf/pull/13348 ?

**Comment by srebhan** [comment]
Fine with me.

**Pull request closed without merging**